project/src/main/kotlin/org/ionproject/core/klass/KlassRepo.kt

```diff
@@ -0,0 +1,71 @@
+package org.ionproject.core.klass
+
+import org.ionproject.core.classSection.ClassSection
+import org.ionproject.core.common.transaction.ITransactionManager
+import org.jdbi.v3.core.transaction.TransactionIsolationLevel
+import org.springframework.stereotype.Component
+
+/**
+ * When the target resource [Class] does not exist, this exception will be thrown.
+ *
+ * This exception <b>shall</b> not be thrown if the [Class] resource does exist, but one or more of
+ * its properties don't (e.g. WAD, 1920v [Class] exists, but has no [ClassSections]).
+ */
+class ClassNotInDbException : Exception()
+
+interface KlassRepo {
+    fun get(id: Int, calendarTerm: String): FullKlass
+    fun getPage(id: Int, page: Int, size: Int): List<Klass>
+}
+
+@Component
+class KlassRepoImplementation(private val tm: ITransactionManager) : KlassRepo {
+
+    /**
+     * Retrieve the target [Class] resource from the database, with all its details.
+     */
+    override fun get(id: Int, calendarTerm: String): FullKlass = tm.run(TransactionIsolationLevel.READ_COMMITTED) { handle ->
+        val klass = handle
+            .createQuery(
+                """select CR.id as cid, CR.acronym, C.term from dbo.Class as C
+                join dbo.Course as CR on C.courseid=CR.id
+                where CR.id=:cid and C.term=:term""".trimIndent())
+            .bind("cid", id)
+            .bind("term", calendarTerm)
```

Comment:
Update https://github.com/i-on-project/core/blob/docs/gh-6-read-api-spec/docs/api/class-sections.md properties to be according with the properties of the returned siren. (CalendarTerm on the return object & class on the read api)

Reply:
I have submitted changes to the API spec via PR, which haven't been approved yet.